### PR TITLE
Fixing and enabling DateTime parse tests with special cultures.

### LINKED ISSUE #3616: Fix Serbian date formats
In the following cultures 

sr-Cyrl-XK
sr-Latn-ME
sr-Latn-RS
sr-Latn-XK

it has wrong date pattern as "d. M. yyyy.". the CLDR don't include spaces. the problem with that is parser get confused between the date and the time separator and fail parse these dates. 
these cultures has time pattern as hh.mm.ss. so we'll have date separator as ". " and time separator as ".". 

The fix is we need to remove the spaces in the date pattern to allow the parser detect such special cases and work. if we cannot fix the format then we'll need to work around this in the parser but this comes with the regression risk.

### PR DESCRIPTION
**Pull request by eerhardt**
These cultures have the same separator character for dates and times.  The tests that were failing on Linux have only a 2-digit year in their short date pattern, which causes problems for DateTime.Parse because it can't tell the difference between a date and a time, so it picks a time and uses the current date.  Fixing the tests to expect this behavior.

Fix #3616 #4599

@tarekgh @ellismg @stephentoub

### CODE REVIEW

File: src/System.Runtime/tests/System/DateTime.cs [hunk 1]
```diff
@@ -362,15 +349,24 @@ internal static void TestDateTimeParsingWithSpecialCultures(string cultureName)
...
-        // truncate the milliseconds as it is not showing in time formatting patterns
-        date = new DateTime(date.Year, date.Month, date.Day, date.Hour, date.Minute, date.Second);
+        DateTime date = new DateTime(2015, 11, 20, 11, 49, 50);
         string dateString = date.ToString(ci.DateTimeFormat.ShortDatePattern, ci);
 
         DateTime parsedDate;
         Assert.True(DateTime.TryParse(dateString, ci, DateTimeStyles.None, out parsedDate));
+        if (ci.DateTimeFormat.ShortDatePattern.Contains("yyyy"))
+        {
         Assert.Equal(date.Date, parsedDate);
```
Comment (tarekgh) [inline src/System.Runtime/tests/System/DateTime.cs]: nit: could you align this source line?

**Comment by tarekgh** [comment]
LGTM

File: src/System.Runtime/tests/System/DateTime.cs (hunk 1 above)
Comment (eerhardt) [inline src/System.Runtime/tests/System/DateTime.cs]: Oops.  Yes, done.

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**